optaplanner-persistence/optaplanner-persistence-jpa/pom.xml

```diff
@@ -58,6 +58,12 @@
       <scope>runtime</scope>
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <!-- Required by JDK9. Configured as 'provided' to avoid the inclusion in the transitive dependency tree. -->
+      <groupId>org.jboss.spec.javax.xml.bind</groupId>
+      <artifactId>jboss-jaxb-api_2.2_spec</artifactId>
+      <scope>provided</scope>
```

Comment:
Are we sure this needs to be scope=provided and not optional=true?
https://stackoverflow.com/questions/5860426/optional-dependency-with-scope-provided-in-maven

Is there a stackoverflow question specifically around jaxb-api being in JDK 8 and not in JDK 9 and how to handle that in maven?

Reply:
I  don't think there is a question about that. My reasoning is the following:

For the JDK8 the jaxb-api is provided (and enabled) by the standard JRE installation, so users don't need to do anything.

For the JDK9 the jaxb-api is by default disabled and users need to provide that themselves (either by depending on the spec jar or by using `--add-modules java.se.ee`)

In case the jaxb-api is actually optional for this module, then we should definitely mark it as such.